components/chatbot/chatController.js

```diff
@@ -41,258 +35,61 @@ function ChatController({
-          <div className="mt-4">
-            <LoadingDots />
-          </div>
-        );
-    }
-  };
-
-  const renderBasedOnSource = (sourceStatus) => {
-    switch (sourceStatus) {
-      case "ChatGPT":
-        return (
-          <div className="flex rounded-lg justify-center items-center max-w-fit bg-white rounded-md">
-            <div className="flex">
-              <PiGlobeSimpleDuotone className="w-6 h-6 mx-4 my-2 text-purple-500" />
-            </div>
-            <div className="my-2 mr-5">
-              <div className="flex items-center">
-                <div className="text-gray text-xs font-bold flex aligns-center">
-                  <span className="mr-2">ChatGPT</span>
-                </div>
-              </div>
-            </div>
-          </div>
-        );
-      case "Google_Search":
-        return (
-          <div className="flex rounded-lg justify-center items-center max-w-fit bg-white rounded-md">
-            <div className="flex">
-              <PiGoogleLogoDuotone className="w-6 h-6 mx-4 my-2 text-red-500" />
-            </div>
-            <div className="my-2 mr-5">
-              <div className="flex items-center">
-                <div className="text-gray text-xs font-bold flex aligns-center">
-                  <span className="mr-2">Google Search</span>
-                </div>
-              </div>
-            </div>
-          </div>
-        );
-      case "Document_QA_System":
-        return (
-          <div className="flex rounded-lg justify-center items-center max-w-fit">
-            <div className="flex">
-              <PiFolderUserDuotone className="w-6 h-6 mx-4 my-2 text-blue-500" />
-            </div>
-            <div className="my-2 mr-5">
-              <div className="flex items-center">
-                <div className="text-gray text-xs font-bold flex aligns-center">
-                  <span className="mr-2">Document Library</span>
-                </div>
-              </div>
-            </div>
-          </div>
-        );
-      case "Document_Display":
-        return (
-          <div className="flex rounded-lg justify-center items-center max-w-fit">
-            <div className="flex">
-              <PiMagnifyingGlassDuotone className="w-6 h-6 mx-4 my-2 text-blue-500" />
-            </div>
-            <div className="my-2 mr-5">
-              <div className="flex items-center">
-                <div className="text-gray text-xs font-bold flex aligns-center">
-                  <span className="mr-2">Document Search</span>
-                </div>
-              </div>
-            </div>
-          </div>
-        );
-      default:
-        return <></>;
+  useEffect(() => {
+    let intervalId;
+    if (isSendChatLoading) {
+      //when you send a message
+      intervalId = setInterval(async () => {
+        if (chatId) {
+          let chatStatus = await getChatStatusAPI(chatId);
+          setResponseStatus(chatStatus);
+        }
+      }, 1000);
```

Comment:
The code block used to be in its parent component which causes unnecessary render.